### PR TITLE
Custom base image definition via meta.yaml.

### PR DESCRIPTION
**Pull request by johanneskoester**
This PR allows to define a custom base image for mulled via
```
extra:
  container:
    base: "<baseimage>"
```

### CODE REVIEW

**Comment by johanneskoester** [comment]
We have to wait for bioconda/bioconda-recipes#4713.

**Comment by johanneskoester** [comment]
@bgruening this is strange: `mulled-build` cannot be found anymore...

**Comment by bgruening** [comment]
Installing it locally seems to work, it is also in the conda tarball ... :(
I try to look at it today.

File: bioconda_utils/pkg_test.py [hunk 1]
```diff
@@ -133,4 +137,8 @@ def test_package(path,
     cmd += channel_args
     cmd += shlex.split(mulled_args)
     logger.debug('mulled-build command: %s' % cmd)
-    return utils.run(cmd)
+
+    env = {}
```
Comment (bgruening) [inline bioconda_utils/pkg_test.py]: You are overwriting here env which is passed to subprocess and this can result in mulled not being on pass, I guess.

Comment (johanneskoester) [inline bioconda_utils/pkg_test.py]: Good catch!

**Comment by johanneskoester** [comment]
@bgruening this looks like a bug in the new mulled-build version: https://travis-ci.org/bioconda/bioconda-utils/jobs/233125954#L1231, am I right?

File: test/test_pkg_test.py [hunk 2]
```diff
@@ -40,16 +58,24 @@ def _build_pkg():
...
-    Make sure calling mulled-build with the wrong arg fails correctly
+    Make sure calling mulled-build with the wrong arg fails correctly.
     """
-    built_package = _build_pkg()
+    built_package = _build_pkg(RECIPE_ONE)
     with pytest.raises(sp.CalledProcessError):
         res = pkg_test.test_package(built_package, mulled_args='--wrong-arg')
+
+
+def test_pkg_test_custom_base_image():
```
Comment (bgruening) [inline test/test_pkg_test.py]: @johanneskoester nearly working. OSX is still broken I guess we need to skip the tests on OSX with `@pytest.fixture(scope='module', params=PARAMS, ids=IDS)`. I just don't understand why this was passing before.

Comment (johanneskoester) [inline test/test_pkg_test.py]: Thanks! I have disabled mulled on osx. Let's see if it works now.

**Comment by johanneskoester** [comment]
Yes, finally fixed! Thanks for your help @bgruening!

**Comment by johanneskoester** [comment]
Now we should be able to fix the click-related recipes. Is it ok if I use the debian container image for them? Or did you find a better solution @bgruening?

**Pull request merged**